openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/DataInsight.spec.ts

```diff
@@ -0,0 +1,274 @@
+
+    for (const data of KPI_DATA) {
+      await page.getByTestId('add-kpi-btn').click();
+
+      await addKpi(page, data);
+    }
+  });
+
+  test('Run DataInsight Application', async ({ page }) => {
+    await settingClick(page, GlobalSettingOptions.APPLICATIONS);
+
+    await page.click(
+      '[data-testid="data-insights-application-card"] [data-testid="config-btn"]'
+    );
+    await page.waitForResponse('**/api/v1/apps/name/DataInsightsApplication?*');
+
+    await page.click('[data-testid="run-now-button"]');
+    await page.waitForResponse(
+      '**/api/v1/apps/trigger/DataInsightsApplication'
+    );
+
+    const { apiContext } = await getApiContext(page);
+
+    await expect
+      .poll(
+        async () => {
+          const response = await apiContext
+            .get(
+              '/api/v1/apps/name/DataInsightsApplication/status?offset=0&limit=1'
+            )
+            .then((res) => res.json());
+
+          return response.data[0].status;
+        },
+        {
+          // Custom expect message for reporting, optional.
+          message: 'Wait for the pipeline to be successful',
+          timeout: 60_000,
+          intervals: [5_000, 10_000],
+        }
+      )
+      .toBe('success');
+
+    // Verify KPI
+
+    await sidebarClick(page, SidebarItem.DATA_INSIGHT);
+
+    await page.waitForSelector('[data-testid="search-dropdown-Team"]');
+    await page.waitForSelector('[data-testid="search-dropdown-Tier"]');
+    await page.waitForSelector('[data-testid="summary-card"]');
+    await page.waitForSelector('[data-testid="kpi-card"]');
+  });
+
+  test('Verifying Data assets tab', async ({ page }) => {
+    await page.waitForResponse(
+      '/api/v1/kpi/playwright-owner-with-percentage-percentage/latestKpiResult'
+    );
+    await page.getByTestId('date-picker-menu').click();
+    await page.getByRole('menuitem', { name: 'Last 60 days' }).click();
+
+    await page
+      .getByTestId('search-dropdown-Team')
+      .waitFor({ state: 'visible' });
+    await page
+      .getByTestId('search-dropdown-Tier')
+      .waitFor({ state: 'visible' });
+    await page.getByTestId('summary-card').waitFor({ state: 'visible' });
+    await page.getByTestId('kpi-card').waitFor({ state: 'visible' });
+    await page
+      .getByTestId('total_data_assets-graph')
+      .waitFor({ state: 'visible' });
+    await page
+      .getByTestId('percentage_of_data_asset_with_description-graph')
+      .waitFor({ state: 'visible' });
+    await page
+      .getByTestId('percentage_of_data_asset_with_owner-graph')
+      .waitFor({ state: 'visible' });
+    await page
+      .getByTestId('percentage_of_service_with_description-graph')
+      .waitFor({ state: 'visible' });
+    await page
+      .getByTestId('percentage_of_service_with_owner-graph')
+      .waitFor({ state: 'visible' });
+    await page
+      .getByTestId('total_data_assets_by_tier-graph')
+      .waitFor({ state: 'visible' });
+  });
```

Comment:
just wanted to know why we are using `.waitFor({ state: 'visible' });` instead of assertion